File: java-soa/inventory-service/src/main/java/com/ecommerce/inventory/infrastructure/web/dto/CreateInventoryRequest.java
package com.ecommerce.inventory.infrastructure.web.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;

public record CreateInventoryRequest(
        @NotNull(message = "Product ID is required")
        UUID productId,

        @NotBlank(message = "Product name is required")
        String productName,

        @Min(value = 0, message = "Quantity must be >= 0")
        int quantityAvailable
) {}

File: dotnet-soa/InventoryService/Infrastructure/Web/Dto/CreateInventoryRequest.cs
using System.ComponentModel.DataAnnotations;

namespace InventoryService.Infrastructure.Web.Dto;

/// <summary>
/// DTO for inventory creation/update requests.
/// </summary>
public record CreateInventoryRequest
{
    [Required]
    public Guid ProductId { get; init; }

    [Required]
    [StringLength(255)]
    public string ProductName { get; init; } = string.Empty;

    [Required]
    [Range(0, int.MaxValue, ErrorMessage = "Quantity must be non-negative")]
    public int Quantity { get; init; }
}
